docs/source/release_notes/release-notes.rst

```diff
@@ -85,6 +85,32 @@ develop
          - Add jitter to backoff on retries to `reduce load <https://www.awsarchitectureblog.com/2015/03/backoff.html>`__ on the server.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1945>`__)
 
+    *    - |fixed|
+         - ``TimestampAllocationFailures`` now correctly propagates ``ServiceNotAvailableException`` if thrown from the timestamp bound store.
+           Previously, a ``NotCurrentLeaderException`` that was thrown from the timestamp store would be wrapped in ``RuntimeException`` before being thrown out, meaning that TimeLock clients saw 500s instead of the intended 503s.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/TBC>`__)
+
+.. <<<<------------------------------------------------------------------------------------------------------------->>>>
+
+
+======
+0.42.2
```

Comment:
Actually, we cut 0.42.2 yesterday which used a different implementation (that defined a new type of exception). This is _not_ in develop and I don't want it to be there - I pushed out something fast for internal deployment, but it looks like it was more complicated than necessary. This fix achieves the same (well, very similar) end with much less code.